### PR TITLE
fix panicks about cgroup on special platforms

### LINKED ISSUE #10897: tikv panicking on startup
### What version of TiKV are you using?

5.2.0

### What operating system and CPU are you using?

Linux Subsystem for Windows, Ubuntu 20.0.4 LTS
As for the CPU info, I have 4 processors each with these specs:
processor       : 0
vendor_id       : GenuineIntel
cpu family      : 6
model           : 61
model name      : Intel(R) Core(TM) i7-5600U CPU @ 2.60GHz
stepping        : 4
microcode       : 0xffffffff
cpu MHz         : 2594.000
cache size      : 256 KB
physical id     : 0
siblings        : 4
core id         : 0
cpu cores       : 2
apicid          : 0
initial apicid  : 0
fpu             : yes
fpu_exception   : yes
cpuid level     : 6
wp              : yes
flags           : fpu vme de pse tsc msr pae mce cx8 apic sep mtrr pge mca cmov pat pse36 clflush dts acpi mmx fxsr sse sse2 ss ht tm pbe syscall nx pdpe1gb rdtscp lm pni pclmulqdq dtes64 monitor ds_cpl vmx smx est tm2 ssse3 fma cx16 xtpr pdcm pcid sse4_1 sse4_2 x2apic movbe popcnt tsc_deadline_timer aes xsave osxsave avx f16c rdrand lahf_lm abm 3dnowprefetch fsgsbase tsc_adjust bmi1 hle avx2 smep bmi2 erms invpcid rtm rdseed adx smap intel_pt ibrs ibpb stibp ssbd
bogomips        : 5188.00
clflush size    : 64
cache_alignment : 64
address sizes   : 36 bits physical, 48 bits virtual
power management:

### Steps to reproduce

I'm not sure what has gone wrong, running "tiup playground" used to work just fine but since a couple days ago whenever I run it TiKV crashes. I have tiup updated to the newest version as well.

### What did you expect?
I expect the instance to start normally, like it used to when I first downloaded TiUP and run "tiup playground"
### What did happened?
Now whenever I try to run it here's what happens:

adam@DESKTOP-N3M2KTC:~$ tiup playground
Starting component `playground`: /home/adam/.tiup/components/playground/v1.5.6/tiup-playground
Using the version v5.2.0 for version constraint "".

If you'd like to use a TiDB version other than v5.2.0, cancel and retry with the following arguments:
    Specify version manually:   tiup playground <version>
    Specify version range:      tiup playground ^5
    The nightly version:        tiup playground nightly

Playground Bootstrapping...
Start pd instance
Start tikv instance
Start tidb instance
tikv quit: exit status 101
thread 'main' panicked at 'assertion failed: ret.contains_key(*controller)', components/tikv_util/src/sys/cgroup.rs:184:9
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
...
check detail log from: /home/adam/.tiup/data/ShtcUUR/tikv-0/tikv.log
Waiting for tidb instances ready
127.0.0.1:4000 ... ⠴

Run with RUST_BACKTRACE=full:
adam@DESKTOP-N3M2KTC:~$ RUST_BACKTRACE=full tiup playground
Starting component `playground`: /home/adam/.tiup/components/playground/v1.5.6/tiup-playground
Using the version v5.2.0 for version constraint "".

If you'd like to use a TiDB version other than v5.2.0, cancel and retry with the following arguments:
    Specify version manually:   tiup playground <version>
    Specify version range:      tiup playground ^5
    The nightly version:        tiup playground nightly

Playground Bootstrapping...
Start pd instance
Start tikv instance
Start tidb instance
Waiting for tidb instances ready
127.0.0.1:4000 ... ⠏
tikv quit: exit status 101
  45:     0x7fb64e00110a - tikv_server::main::h64f2cb4cacd69b59
                               at /home/jenkins/agent/workspace/optimization-build-tidb-linux-amd/go/src/github.com/pingcap/tikv/cmd/tikv-server/src/main.rs:159:22
  46:     0x7fb64dfea373 - core::ops::function::FnOnce::call_once::he841198ab39dc2ae
                               at /rustc/2faabf579323f5252329264cc53ba9ff803429a3/library/core/src/ops/function.rs:227:5
  47:     0x7fb64dfea373 - std::sys_common::backtrace::__rust_begin_short_backtrace::h8774cf5834ef8cd1
                               at /rustc/2faabf579323f5252329264cc53ba9ff803429a3/library/std/src/sys_common/backtrace.rs:125:18
  48:     0x7fb64e003824 - main
  49:     0x7fb64d7070b3 - __libc_start_main
  50:     0x7fb64dfe9487 - <unknown>
  51:                0x0 - <unknown>
Waiting for tidb instances ready
127.0.0.1:4000 ... Error
To view the dashboard: http://127.0.0.1:2379/dashboard
PD client endpoints: [127.0.0.1:2379]
To view the Prometheus: http://127.0.0.1:9090
To view the Grafana: http://127.0.0.1:3000

I am guessing that this is a problem with something on my machine rather than a bug in the code itself, but I'm not sure what it is. I've tried uninstalling and reinstalling tiup, but it still won't work. I'm confused because it worked as recently as a few days ago, and then stopped working, but I don't know of anything out of the ordinary that could have made it start crashing. Can anyone help me out?

### PR DESCRIPTION
**Pull request by hicqu**
Signed-off-by: qupeng <qupeng@pingcap.com>

### What problem does this PR solve?
Fix some panicks about cgroup on special platfoms(e.g. WSL, containers).

Issue Number: close #10897 and #10889 .

### What is changed and how it works?

* allow `/sys/fs/cgroup` not exists
* handle relative path in the mountinfo file correctly
* all sub systems are not required to be supported for containers

### Related changes

- PR to update `pingcap/docs`/`pingcap/docs-cn`:
- PR to update `pingcap/tidb-ansible`:
- Need to cherry-pick to the release branch

### Check List 

Tests 

- Unit test

### Release note 

```release-note
None
```

### CODE REVIEW

File: components/tikv_util/src/sys/cgroup.rs [hunk 1]
```diff
@@ -58,85 +59,87 @@ pub struct CGroupSys {
 }
 
 impl CGroupSys {
-    pub fn new() -> Self {
+    pub fn new() -> Result<Self, String> {
         let lines = read_to_string("/proc/self/cgroup").unwrap();
-        let is_v2 = is_cgroup2_unified_mode();
+        let is_v2 = is_cgroup2_unified_mode()?;
```
Comment (sticnarf) [inline components/tikv_util/src/sys/cgroup.rs:64]: This change does not seem to make any difference. If this returns an error, we are using a default `CGroupSys`. Then, I think later calls like `cpuset_cores` will still panic?

Comment (hicqu) [reply components/tikv_util/src/sys/cgroup.rs:64]: The new test case `test_defult_cgroup_sys` shows that a default `CGroupSys` will return default value for `cpuset_cores` and other methods.

Comment (sticnarf) [reply components/tikv_util/src/sys/cgroup.rs:64]: Ah, get it.

**Review by sticnarf — APPROVED** [review]
LGTM

**Review by hunterlxt — APPROVED** [review]
LGTM

**Pull request merged**